octopusdeploy_framework/resource_project_versioning_strategy.go

```diff
@@ -164,27 +169,32 @@ func (r *projectVersioningStrategyResource) Delete(ctx context.Context, req reso
 }
 
 func mapStateToProjectVersioningStrategy(state *schemas.ProjectVersioningStrategyModel) *projects.VersioningStrategy {
-	var donorPackageStepID *string
-	donorPackageStepIDString := state.DonorPackageStepID.ValueString()
-	if donorPackageStepIDString != "" {
-		donorPackageStepID = &donorPackageStepIDString
+	projectVersioningStrategy := &projects.VersioningStrategy{}
+	if !(state.Template.IsNull()) {
+		projectVersioningStrategy.Template = state.Template.ValueString()
 	}
-
-	return &projects.VersioningStrategy{
-		Template:           state.Template.ValueString(),
-		DonorPackageStepID: donorPackageStepID,
-		DonorPackage: &packages.DeploymentActionPackage{
+	if !(state.DonorPackageStepID.IsNull()) {
+		projectVersioningStrategy.DonorPackageStepID = state.DonorPackageStepID.ValueStringPointer()
+	}
+	if !(state.DonorPackage == nil) {
```

Comment:
I am guessing you do this to match the above line over != nil?

Reply:
Yeah, consistency of... polarity?... of the check, so that it's easy to scan down and understand. I would've preferred an `IsNull()` method, but it doesn't exist on the `DonorPackage` struct versus the String above.